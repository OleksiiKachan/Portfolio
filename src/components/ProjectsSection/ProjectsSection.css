.projectsSection {
    padding: 30px 5%;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: stretch;
}

.projectsSection__header {
    align-self: center;
    margin-bottom: 25px;
}

.projectsSection__projectsFilter {
	list-style-type: none;
	display: flex;
	justify-content: center;
	align-items: center;
	margin-bottom: 25px;
}

.projectsSection__projectsFilterItem {
    margin: 0 10px;
	padding: 10px;
	cursor: pointer;
}

.projectsSection__projectsFilterItem_clicked {
    border-bottom: 2px solid #0f82f9;
}

.projectsSection__projectsFilterButton {
	background: none;
	border: none;
	font-size: 10pt;
	font-weight: 700;
	color: #0f82f9;
}

.projectsSection__content {
    display: flex;
	justify-content: center;
    align-items: flex-start;
    flex-wrap: wrap
}