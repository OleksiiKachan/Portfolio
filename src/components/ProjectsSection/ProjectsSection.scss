@import '../../assets/style/keyframes';

.projectsSection {
  padding: 30px 5%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: stretch;
}

.projectsSection__header {
  align-self: center;
  margin-bottom: 25px;
}

.projectsSection__content {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-wrap: wrap;
}
