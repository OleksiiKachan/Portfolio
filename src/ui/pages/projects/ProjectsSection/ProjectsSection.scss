@import '../../../assets/style/mixins';

.projectsSection {
  padding: 30px 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: stretch;

  &__header {
    align-self: center;
    margin-bottom: 25px;
  }

  &__content {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 1fr 1fr;
    grid-column-gap: 0px;
    grid-row-gap: 0px;

    @include respond(tablet) {
      display: flex;
      flex-direction: column;
      justify-content: stretch;
    }
  }
}
