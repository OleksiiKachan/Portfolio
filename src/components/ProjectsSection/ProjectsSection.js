import React, { Component } from 'react';
import classNames from 'classnames';
import './ProjectsSection.css';
import ProjectCard from '../ProjectCard';

/*
    <ProjectsSection projects={} projectFilterMethod={}/>
*/
class ProjectsSection extends Component {
    constructor(props) {
        super(props);

        this.state = {
            previousProjectType: 'all',
            projectType: 'all',
            projects: this.props.projects
        };
    }    

    setProjectType = (newProjectType) => {
        this.setClickedFilterButton(newProjectType);
        this.setState({
            previousProjectType: this.state.projectType,
            projectType: newProjectType,
            projects: this.filterProjects(newProjectType)
        });
    } 

    filterProjects = (projectType) => {
        return projectType === 'all' ? this.props.projects : this.props.projects.filter((project) => { return project.projectType === projectType})

    }

    componentDidUpdate = () => {
        this.setClickedFilterButton();
    }

    setClickedFilterButton = () => {
        const oldClickedButton = document.getElementById(`projectsFilterButton_${this.state.previousProjectType}`);
        oldClickedButton.parentElement.classList.remove('projectsSection__projectsFilterItem_clicked');
        const newClickedButton = document.getElementById(`projectsFilterButton_${this.state.projectType}`);
        newClickedButton.parentElement.classList.add('projectsSection__projectsFilterItem_clicked');
    }

    render() {
        return (
            <div id='portfolio' className={classNames('projectsSection', this.props.className)}>
                <h2 className={classNames('projectsSection__header')}>Portfolio</h2>
                <ul className={classNames('projectsSection__projectsFilter')}>
                    <li className={classNames('projectsSection__projectsFilterItem', 'projectsSection__projectsFilterItem_clicked')}>
                        <button id='projectsFilterButton_all' onClick={() => this.setProjectType('all')} className={classNames('projectsSection__projectsFilterButton')}>All Projects</button>
                    </li>
                    <li className={classNames('projectsSection__projectsFilterItem')}>
                        <button id='projectsFilterButton_development' onClick={() => this.setProjectType('development')} className={classNames('projectsSection__projectsFilterButton')}>Development</button>
                    </li>
                    <li className={classNames('projectsSection__projectsFilterItem')}>
                        <button id='projectsFilterButton_design' onClick={() => this.setProjectType('design')} className={classNames('projectsSection__projectsFilterButton')}>Design</button>
                    </li>
                </ul>
                <div className={classNames('projectsSection__content')}>
                {
                    this.state.projects.map(project => {
                        return(
                            <ProjectCard project={project}/>
                        );
                    })
                }
                </div>
            </div>
        );
    }
}

export default ProjectsSection;