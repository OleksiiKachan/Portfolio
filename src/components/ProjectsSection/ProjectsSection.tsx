import React, { Component } from 'react';
import classNames from 'classnames';
import { ProjectType } from '../../lib/types';
import ProjectCard from './ProjectCard';
import ProjectsFilter from './ProjectsFilter/ProjectsFilter';
import './ProjectsSection.scss';

/*
    <ProjectsSection
      projects={} 
      className=''
    />
*/

type PropsType = {
  projects: Array<ProjectType>;
  className?: string;
};

type StateType = {
  previousProjectType: string;
  projectType: string;
  projects: Array<ProjectType>;
};

export default class ProjectsSection extends Component<PropsType, StateType> {
  constructor(props: PropsType) {
    super(props);

    this.state = {
      previousProjectType: 'all',
      projectType: 'all',
      projects: this.props.projects,
    };

    this.setProjectType = this.setProjectType.bind(this);
  }

  setProjectType = (newProjectType: string) => {
    this.setClickedFilterButton();
    this.setState({
      previousProjectType: this.state.projectType,
      projectType: newProjectType,
      projects: this.filterProjects(newProjectType),
    });
  };

  filterProjects = (projectType: string) => {
    return projectType === 'all'
      ? this.props.projects
      : this.props.projects.filter((project: { projectType: string }) => {
          return project.projectType === projectType;
        });
  };

  componentDidUpdate = () => {
    this.setClickedFilterButton();
  };

  setClickedFilterButton = () => {
    const oldClickedButton = document.getElementById(
      `projectsFilterButton_${this.state.previousProjectType}`
    );
    oldClickedButton!.parentElement!.classList.remove(
      'projectsSection__projectsFilterItem_clicked'
    );
    const newClickedButton = document.getElementById(
      `projectsFilterButton_${this.state.projectType}`
    );
    newClickedButton!.parentElement!.classList.add(
      'projectsSection__projectsFilterItem_clicked'
    );
  };

  render() {
    return (
      <div
        id="portfolio"
        className={classNames('projectsSection', this.props.className)}
      >
        <h2 className={classNames('projectsSection__header')}>Projects</h2>

        <ProjectsFilter
          activeType={this.state.projectType}
          onClick={this.setProjectType}
        />
        <div className={classNames('projectsSection__content')}>
          {this.state.projects.map((project: ProjectType) => {
            return (
              <ProjectCard
                key={this.state.projects.indexOf(project)}
                id={project.id}
                title={project.title}
                stack={project.shortStack}
                cardImage={project.cardImage}
              />
            );
          })}
        </div>
      </div>
    );
  }
}
