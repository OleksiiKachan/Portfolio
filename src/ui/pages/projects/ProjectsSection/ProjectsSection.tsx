import React from 'react';
import classNames from 'classnames';
import { ProjectType } from '../../../../lib/types';
import ProjectCard from './ProjectCard/ProjectCard';
import './ProjectsSection.scss';

/*
    <ProjectsSection
      projects={} 
      className=''
    />
*/

type PropsType = {
  projects: Array<ProjectType>;
  className?: string;
};

export default ({ projects, className }: PropsType) => {
  return (
    <div id="portfolio" className={classNames('projectsSection', className)}>
      <h2 className={classNames('projectsSection__header')}>Projects</h2>
      <div className={classNames('projectsSection__content')}>
        {projects.map((project: ProjectType) => {
          const index = projects.indexOf(project);
          return (
            <ProjectCard
              key={index}
              id={project.id}
              title={project.title}
              stack={project.shortStack}
              cardImage={project.cardImage}
              description={project.shortDescription}
              settings={project.settings}
              style={{
                transform: `${
                  index % 2 === 0 ? `translateX(-100%)` : `translateX(100%)`
                }`,
                animation: `${
                  index % 2 === 0
                    ? `fromLeftToCenter-noPop`
                    : `fromRightToCenter-noPop`
                } 1s ease-in-out forwards`,
                animationDelay: `${(index / 1.5) * 100}ms`,
              }}
            />
          );
        })}
      </div>
    </div>
  );
};
