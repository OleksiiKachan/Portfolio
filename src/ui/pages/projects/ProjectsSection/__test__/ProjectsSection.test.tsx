import React from 'react';
import { shallow } from 'enzyme';
import ProjectsSection from '../ProjectsSection';

describe('OutlineButton', () => {
  it('Snapshot', () => {
    const component = shallow(
      <ProjectsSection
        projects={[
          {
            projectType: 'development',
            title: 'AGW MES / WMS for automotive industry',
            shortStack: '.NET Framework',
            coverImage: 'https://picsum.photos/4212/2768',
            description:
              'While working in OLSOM company I worked with some of the biggest automotive manufacturers. I was developing new features for AGW system (which is the MES/WMS for manufacturing). Also I did deployment on new manufactories and support tickets.',
          },
          {
            projectType: 'development',
            title: 'AGW MES / WMS for automotive industry',
            shortStack: '.NET Framework',
            coverImage: 'https://picsum.photos/4212/2768',
            description:
              'While working in OLSOM company I worked with some of the biggest automotive manufacturers. I was developing new features for AGW system (which is the MES/WMS for manufacturing). Also I did deployment on new manufactories and support tickets.',
          },
        ]}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
