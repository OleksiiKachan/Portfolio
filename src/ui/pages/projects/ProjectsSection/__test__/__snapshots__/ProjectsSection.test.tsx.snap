// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton Snapshot 1`] = `
<div
  className="projectsSection"
  id="portfolio"
>
  <h2
    className="projectsSection__header"
  >
    Projects
  </h2>
  <div
    className="projectsSection__content"
  >
    <_default
      key="0"
      stack=".NET Framework"
      style={
        Object {
          "animation": "fromLeftToCenter-noPop 1s ease-in-out forwards",
          "animationDelay": "0ms",
          "transform": "translateX(-100%)",
        }
      }
      title="AGW MES / WMS for automotive industry"
    />
    <_default
      key="1"
      stack=".NET Framework"
      style={
        Object {
          "animation": "fromRightToCenter-noPop 1s ease-in-out forwards",
          "animationDelay": "66.66666666666666ms",
          "transform": "translateX(100%)",
        }
      }
      title="AGW MES / WMS for automotive industry"
    />
  </div>
</div>
`;
